### DISCUSSION TITLE
[V5] Invalid default value for 'created_at'

### DISCUSSION THREAD

**Discussion by skoenfaelt** [adonisjs/core#2002] [Help] [answered]
Hello,
I followed the instructions of https://preview.adonisjs.com/guides/auth/introduction with MySQL as database driver and api guard.

At https://preview.adonisjs.com/guides/auth/setup#run-migrations i run into an error. First I needed to fix the default value for the timestamps on the users migration.

Then I rerun `node ace migration:run` but now I run into a different error
**ER_INVALID_DEFAULT: Invalid default value for 'created_at'**
This comes from the _api_tokens_ migration

      table.timestamp('created_at', { useTz: true }).notNullable()

I cant find any solutions on google according that problem.

**Comment by skoenfaelt** [comment]
I found a solution on that, whicht should be done by default.

at _(api_tokens migrationfile)_
`  table.timestamp('created_at', { useTz: true }).notNullable()`
change it to

`  table.timestamp('created_at', { useTz: true }).notNullable().defaultTo(this.now())`

**Comment by dawidow** [comment, marked as answer]
If you use MySQL 5.x you should remove all parameters from timestamps `table.timestamp()` or pass 2 true values `table.timestamp(true, true)`. In MySQL 8.x you don't need to do anything. It should be fine.
